app/controllers/application.js

```diff
@@ -1,214 +1,12 @@
-import { later } from "@ember/runloop";
-import { on } from "@ember/object/evented";
-import { computed } from "@ember/object";
-import { alias } from "@ember/object/computed";
-import { inject as service } from "@ember/service";
-import Controller from "@ember/controller";
-import { getOwner } from "@ember/application";
-import { containsAny } from "../utils/helpers";
-import config from "../config/environment";
-import AjaxPromise from "browse/utils/ajax-promise";
-import cancelOrderMixin from "../mixins/cancel_order";
-import quantityUpdateMixin from "browse/mixins/quantity_update";
-import _ from "lodash";
+import applicationController from "./application_root";
 
-export default Controller.extend(cancelOrderMixin, quantityUpdateMixin, {
-  isMobileApp: config.cordova.enabled,
-  appVersion: config.APP.VERSION,
-  subscription: service(),
-  screenresize: service(),
-  messageBox: service(),
-  orderService: service(),
-  cart: service(),
-  i18n: service(),
-  showSidebar: true,
-  isWideScreen: alias("screenresize.isWideScreen"),
-  isHomePage: computed("currentPath", function() {
-    return this.get("currentPath") === "home";
-  }),
-  app_id: config.APP.ANDROID_APP_ID,
-  ios_app_id: config.APP.APPLE_APP_ID,
-  appTitle: config.APP.TITLE,
-  bannerImage: config.APP.BANNER_IMAGE,
-  bannerReopenDays: config.BANNER_REOPEN_DAYS,
+export default applicationController.extend({
+  init() {
+    this._super(...arguments);
 
-  initializer: on("init", function() {
-    this.get("subscription").wire();
```

Comment:
@swatijadhav Is the point of all of this to prevent `wire()` from being called for all instances of application controller ?

If so this probably not the right way to fix this, you can simply add the `if (this.isSubscribed) return` statement inside the subscription service's wire method (services are instantiated only once and shared everywhere)

Also, I don't think there's a good reason for all those controllers to inherit from `applicationController`, we can probably remove that. The application controller can be injected into other controllers if they need to call some methods against it

Cutting the controller in half with `application.js` and `application_root.js` is too hacky